workflow/libero_flow_poc/libero_flow/worker.py

```diff
@@ -0,0 +1,157 @@
+import json
+import logging
+import time
+from typing import Any, Dict
+
+import pika
+from pika.exceptions import (
+    ChannelClosed,
+    ConnectionClosed,
+    IncompatibleProtocolError
+)
+from libero_flow.conf import (
+    ACTIVITY_RESULT_EXCHANGE,
+    ACTIVITY_RESULT_QUEUE,
+    SCHEDULED_ACTIVITY_QUEUE,
+)
+from libero_flow.flow_loader import FlowLoader
+from libero_flow.event_utils import (
+    get_base_message,
+    get_channel,
+    message_handler,
+    setup_exchanges_and_queues,
+    DELIVERY_MODE_PERSISTENT,
+)
+from libero_flow.session_store import get_session
+from libero_flow.state_utils import (
+    get_activity_state,
+    send_workflow_event,
+    update_activity_status,
+    WORKFLOW_ACTIVITY_STARTED,
+    WORKFLOW_ACTIVITY_FINISHED,
+)
+
+
+logger = logging.getLogger(__name__)
+fh = logging.FileHandler(f'{__name__}.log')
+logger.addHandler(fh)
+
+
+def run_activity(activity_id: str) -> Dict:
+    """Execute target `Activity` class.
+
+    :param activity_id: str
+    :return: dict
+    """
+    try:
+        result = {
+            'activity_id': activity_id,
+            'result': ''
+        }
+
+        update_activity_status(activity_id, status='In Progress')
+
+        loader = FlowLoader()
+        activity_state = get_activity_state(activity_id=activity_id)
+        activity_class = loader.get_activity(activity_state['name'])
+
+        if activity_class:
+            send_workflow_event(workflow_id=activity_state['workflow'],
+                                event_type=WORKFLOW_ACTIVITY_STARTED)
```

Comment:
these events should definitely contain `activity_class` or some normalization of it

Reply:
I agree. It should probably be a generic field, something like 'info' as it's not just activity based events being logged, but also workflow and decision events.

It would require touching a few parts (the db model, api, the func calls) so is probably only worth touching if we are to expand on this experiment past this PR.